### PR TITLE
Pull Request

### PR DESCRIPTION
**Pull request by kirannonstop**
1. Session maintained and cleared
2. Firestore connection added
3. Logout button added
4. Added and Retrieved user on click of LeaderBoard button
5. Kept old google-signin-client_id

### CODE REVIEW

File: lib/data/services/auth/sp_service.dart [hunk 1]
```diff
@@ -1,13 +1,23 @@
 import 'package:shared_preferences/shared_preferences.dart';
 
+import '../../../ui/utils/constant.dart';
```
Comment (ProjectAJ14) [inline lib/data/services/auth/sp_service.dart]: Why do you need `SharedPreferences` ? `firebaseAuth` should be sufficient

File: lib/data/services/firestore/firestore_service.dart [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+          Constants.score: user.score,
+        })
+        .then((value) => debugPrint("User added successfully!"))
+        .catchError((error) => debugPrint('Error adding user $error'));
+  }
+
+  Future<AppResponse> getAllTopScoredUsers() async {
+    try {
+      AppLoader.show();
+      QuerySnapshot querySnapshot = await users.get();
```
Comment (ProjectAJ14) [inline lib/data/services/firestore/firestore_service.dart]: Use `where` query here

File: lib/data/services/firestore/firestore_service.dart [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      List<User> userList = [];
+      List<User> top10UserList = [];
+      for (var doc in querySnapshot.docs) {
+        final data = doc.data() as Map<String, dynamic>;
+
+        userList.add(
```
Comment (ProjectAJ14) [inline lib/data/services/firestore/firestore_service.dart]: No need of this code

File: lib/data/services/firestore/firestore_service.dart [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import 'package:ns_utils/extensions/map.dart';
+
+import '../../../ui/utils/app_loader.dart';
+import '../../../ui/utils/constant.dart';
+import '../../models/app_response.dart';
+import '../../models/user.dart';
+
+FireStoreService get fireStore => Get.find<FireStoreService>();
+
+class FireStoreService {
```
Comment (ProjectAJ14) [inline lib/data/services/firestore/firestore_service.dart]: This should be a `repo` not service

File: web/index.html [hunk 5]
```diff
@@ -24,7 +24,9 @@
     the `--base-href` argument provided to `flutter build`.
   -->
   <meta name="google-signin-client_id" content="820454982989-rhebl9qcf7m5jslm3u77gkuphkef4jhi.apps.googleusercontent.com">
+  <!-- <meta name="google-signin-client_id" content="199053750907-tjb2858s21810qvtkdrjipffvc1nid1l.apps.googleusercontent.com"> -->
```
Comment (ProjectAJ14) [inline web/index.html]: remove commented code

File: lib/ui/utils/constant.dart [hunk 6]
```diff
@@ -1 +1,7 @@
-
+class Constants {
+  static const email = "email";
+  static const displayName = "display_name";
+  static const score = "score";
+  static const isAuthenticated = "isAuthenticated";
```
Comment (ProjectAJ14) [inline lib/ui/utils/constant.dart]: Use snake_case

File: lib/ui/screens/top_users/top_users_screen.dart [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+        children: [
+          setTextValue(user.displayName, false),
+          setTextValue(user.email, false),
+          setTextValue(user.score.toString(), false),
+        ],
+      ),
+    );
+  }
+
+  Widget setTextValue(String strName, bool columnHeader) {
```
Comment (ProjectAJ14) [inline lib/ui/screens/top_users/top_users_screen.dart]: Extracting widgets to a method is considered as a Flutter anti-pattern

File: lib/ui/screens/top_users/top_users_screen.dart [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+        children: [
+          setTextValue("Name", true),
+          setTextValue("Email", true),
+          setTextValue("Score", true),
+        ],
+      ),
+    );
+  }
+
+  Widget setValues(User user) {
```
Comment (ProjectAJ14) [inline lib/ui/screens/top_users/top_users_screen.dart]: Extracting widgets to a method is considered as a Flutter anti-pattern

File: lib/ui/screens/top_users/top_users_screen.dart [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+import '../../utils/constant.dart';
+
+class TopUserScreen extends StatelessWidget {
+  const TopUserScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (context, snapshot) {
+        var quarterTurns = 0;
```
Comment (ProjectAJ14) [inline lib/ui/screens/top_users/top_users_screen.dart:16]: Why?

**Review by ProjectAJ14 — CHANGES_REQUESTED** [review]

File: web/index.html (hunk 5 above)
Comment (kirannonstop) [reply web/index.html]: Removed

File: lib/ui/screens/top_users/top_users_screen.dart (hunk 9 above)
Comment (kirannonstop) [reply lib/ui/screens/top_users/top_users_screen.dart:16]: To rotate the screen when window size minimizes

File: lib/data/services/firestore/firestore_service.dart (hunk 2 above)
Comment (kirannonstop) [reply lib/data/services/firestore/firestore_service.dart]: await users
          .orderBy(Constants.score, descending: true)
          .limit(10)
          .get()
          Added like this, get top 10 users

File: lib/data/services/firestore/firestore_service.dart (hunk 3 above)
Comment (kirannonstop) [reply lib/data/services/firestore/firestore_service.dart]: Yes removed, after providing an order clause no need for this code

File: lib/data/services/firestore/firestore_service.dart (hunk 4 above)
Comment (kirannonstop) [reply lib/data/services/firestore/firestore_service.dart]: Done

File: lib/ui/utils/constant.dart (hunk 6 above)
Comment (kirannonstop) [reply lib/ui/utils/constant.dart]: Done

File: lib/ui/screens/top_users/top_users_screen.dart (hunk 8 above)
Comment (kirannonstop) [reply lib/ui/screens/top_users/top_users_screen.dart]: Made it as a stateless widget class now.

File: lib/ui/screens/top_users/top_users_screen.dart (hunk 7 above)
Comment (kirannonstop) [reply lib/ui/screens/top_users/top_users_screen.dart]: Made it as a stateless widget class now.

File: lib/data/services/auth/sp_service.dart (hunk 1 above)
Comment (kirannonstop) [reply lib/data/services/auth/sp_service.dart]: Done

File: lib/app/services/auth/auth_service.dart [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import '../../../data/models/app_response.dart';
+
+const noCredentialsWereFound = 'No credentials were found';
```
Comment (rajan-nonstopio) [inline lib/app/services/auth/auth_service.dart]: Move string to constants file

File: lib/ui/routes/route_constants.dart [hunk 11]
```diff
@@ -5,4 +5,5 @@ class RouteConstants {
   static const String splashScreen = "/";
   static const String signIn = "/sign_in";
   static const String home = "/home";
+  static const String topUsers = "/topUsers";
```
Comment (rajan-nonstopio) [inline lib/ui/routes/route_constants.dart]: Keep a similar formate for all route names.
Change String to "/top_users"

File: lib/ui/screens/top_users/top_users_screen.dart (hunk 9 above)
Comment (rajan-nonstopio) [reply lib/ui/screens/top_users/top_users_screen.dart:16]: Should we make landscape the default for mobile devices?

File: lib/app/services/auth/auth_service.dart (hunk 10 above)
Comment (kirannonstop) [reply lib/app/services/auth/auth_service.dart]: Done

File: lib/ui/routes/route_constants.dart (hunk 11 above)
Comment (kirannonstop) [reply lib/ui/routes/route_constants.dart]: Done

File: lib/ui/screens/top_users/top_users_screen.dart (hunk 9 above)
Comment (kirannonstop) [reply lib/ui/screens/top_users/top_users_screen.dart:16]: As of now it is made for Web app, not checked for platform

**Comment by kirannonstop** [comment]
Hi @ProjectAJ14 , Kindly check [this](https://www.dropbox.com/scl/fi/hpe7qlhgw5ic9iw7upicn/LeaderboardVideoDemo.webm?rlkey=03dwq6lw1g0d7kglnrxdr4cig&dl=0) video.

**Pull request merged**